config/v1/types_node.go

```diff
@@ -46,6 +46,17 @@ type NodeSpec struct {
 	// the status and corresponding reaction of the cluster
 	// +optional
 	WorkerLatencyProfile WorkerLatencyProfileType `json:"workerLatencyProfile,omitempty"`
+
+	// MinimumKubeletVersion is the lowest version of a kubelet that can meaningfully join the cluster.
+	// Specifically, the apiserver will deny most authorization requests of kubelets that are older
+	// than the specified version, only allowing the kubelet to get and update its node object, and perform
+	// subjectaccessreviews.
```

Comment:
This is interesting technical detail, but, what does this mean to an end user?

Reply:
good question, I added it from @enxebre 's review, I'm not feeling too opinionated on keeping it

Reply:
I think Alberto's question was fair, I'm just not sure we are adding the right context

What does it look like to an end user when the kubelet is having these various authzs fail? 

Does it mean pods won't schedule there? Does it mean the node goes not ready? Is the node actually still functional?

Reply:
how does 
```
	// minimumKubeletVersion is the lowest version of a kubelet that can join the cluster.
	// Specifically, the apiserver will deny most authorization requests of kubelets that are older
	// than the specified version, only allowing the kubelet to get and update its node object, and perform
	// subjectaccessreviews.
	// This means the kubelet won't be able to view API objects it's responsible for running,
	// and will eventually be marked as NotReady.

``` 
work for you?

Reply:
that should be blocked at admission: https://github.com/openshift/kubernetes/pull/2104/files#diff-5c437405eeba0789e9c42802e3f36cf6bdafd59d6d5c5dbaa6b66a2e02948bd7R117-R122 
So really this will only affect new kubelets that attempt to join the cluster after the min version is in place. those kubelets won't be able to run anything

Reply:
So perhaps then we take what you had above with a slight tweak?
```
// This means, any kubelet that attempts to join the cluster will not be able to run any assigned workloads, and will eventually be marked as not ready.
```

When you say will eventually be marked not ready, surely it will never get ready, since the CNI won't be able to initialise the network right?

Reply:
yeah unless it manages to upgrade itself which would require manual intervetion

Reply:
updated to your suggestion